feat(dashboard): Reduce onboarding friction NV-6238

iampearceman opened the pull request.

iampearceman committed:
feat(dashboard): onboarding flow improvement

iampearceman committed:
fixes

iampearceman committed:
Merge branch 'next' into onboarding-emil

iampearceman committed:
Added tracking

iampearceman committed:
Merge branch 'onboarding-emil' of https://github.com/novuhq/novu into onboarding-emil

scopsy commented on apps/dashboard/src/utils/routes.ts:
```diff
@@ -2,7 +2,7 @@ export const ROUTES = {
   SIGN_UP: '/auth/sign-up',
   SIGNUP_ORGANIZATION_LIST: '/auth/organization-list',
-  SIGNUP_QUESTIONNAIRE: '/onboarding/questionnaire',
+  // SIGNUP_QUESTIONNAIRE: '/onboarding/questionnaire', // In attempt to reduce onboarding flow and got users to onboard faster. We decided to comment this out for now.
```
No need to comment, just remove this.

iampearceman replied:
Done.

scopsy commented on apps/dashboard/src/pages/inbox-usecase-page.tsx:
```diff
@@ -1,23 +1,262 @@
 
+  const shouldShowLoading = !requiredData || delightfulState.phase !== 'ready';
+
+  useEffect(() => {
+    if (shouldShowLoading) {
+      const refreshTimeout = setTimeout(() => {
+        window.location.reload();
+      }, 3000);
+
+      return () => clearTimeout(refreshTimeout);
+    }
+  }, [shouldShowLoading]);
```
What his is? Or why it's needed?

iampearceman replied:
I thought that yes - be we will go with a different strategy

iampearceman replied:
I've removed it.

scopsy commented on apps/dashboard/src/pages/inbox-usecase-page.tsx:
```diff
@@ -1,23 +1,262 @@
+          2000 + newCount * 1000
+        );
+      } else {
+        setTimeout(() => {
+          window.location.reload();
+        }, 3000);
```
Why is t his one needed?

iampearceman replied:
Every time when trying to load the initial state for some reason it just sticks, and usually a hard refresh enables the loading of all the values and variables that are needed. I haven't found another way to create this experience smoother so far.

iampearceman replied:
I've removed it.

scopsy commented on apps/dashboard/src/components/welcome/inbox-connected-guide.tsx:
```diff
@@ -1,23 +1,54 @@
   const { triggerWorkflow, isPending } = useTriggerWorkflow(environment);
   useInitDemoWorkflow(environment);
 
+  const curlCommand = `curl -X POST 'https://api.novu.co/v1/events/trigger' \\
```
this should be adjusted to eu aswell.

scopsy commented on apps/dashboard/src/components/welcome/inbox-connected-guide.tsx:
```diff
@@ -1,23 +1,54 @@
   useInitDemoWorkflow(environment);
 
+  const curlCommand = `curl -X POST 'https://api.novu.co/v1/events/trigger' \\
+  -H 'Authorization: ApiKey YOUR_NOVU_SECRET_KEY' \\
```
Shouldn't this contain the secret key here aswell?

iampearceman replied:
Now it does.

scopsy commented on apps/dashboard/src/components/primitives/sonner-helpers.tsx:
```diff
@@ -2,6 +2,15 @@ import { ReactNode } from 'react';
+  style: {
+    left: '50%',
+    transform: 'translateX(-50%)',
+  },
```
Why is this needed? bottom-center should be enough for positioning

scopsy commented on apps/dashboard/src/components/auth/inbox-preview-content.tsx:
```diff
@@ -56,95 +40,63 @@ export function InboxPreviewContent({
-          </Inbox>
-        </div>
-      )}
+      <style>
+        {`
+          .nt-relative.nt-flex.nt-shrink-0.nt-flex-col.nt-justify-center.nt-items-center.nt-gap-1.nt-mt-auto.nt-py-3.nt-text-foreground-alpha-400 {
+            display: none !important;
+          }
+          .inbox-scroll-container {
+            height: 500px;
+            overflow-y: auto;
+            overflow-x: hidden;
+            /* Hide scrollbar for Chrome, Safari and Opera */
+            scrollbar-width: none; /* Firefox */
+            -ms-overflow-style: none; /* Internet Explorer 10+ */
+          }
+          .inbox-scroll-container::-webkit-scrollbar {
+            display: none; /* Hide scrollbar for Chrome, Safari and Opera */
+          }
+          /* Hide scrollbars for all elements within the inbox */
+          .inbox-scroll-container * {
+            scrollbar-width: none; /* Firefox */
+            -ms-overflow-style: none; /* Internet Explorer 10+ */
+          }
+          .inbox-scroll-container *::-webkit-scrollbar {
+            display: none; /* Hide scrollbar for Chrome, Safari and Opera */
+          }
+        `}
+      </style>
```
Why are we using inline styles here not tailwind?

iampearceman replied:
Using tailwind now

scopsy commented on apps/dashboard/src/components/auth/inbox-playground.tsx:
```diff
@@ -1,222 +1,206 @@
+  totalSteps: 4,
+} as const;
+
+function generateCurlCommand(userId: string): string {
+  return `curl -X POST ${window.location.origin}/api/v1/events/trigger \\
+  -H "Content-Type: application/json" \\
+  -H "Authorization: Bearer YOUR_API_KEY" \\
+  -d '{
+    "name": "${ONBOARDING_DEMO_WORKFLOW_ID}",
+    "to": {
+      "subscriberId": ${JSON.stringify(userId)}
+    },
+    "payload": {}
+  }'`;
```
I thin i've already seen this snippet somewhere in the this PR, let's make sure it's reusable

scopsy commented on apps/dashboard/src/components/auth/customize-inbox-playground.tsx:
```diff
@@ -4,7 +4,7 @@ import { RiLayoutLine } from 'react-icons/ri';
 import { ColorPicker } from '@/components/primitives/color-picker';
 
-import type { InboxPlaygroundFormData } from './inbox-playground';
+// import type { InboxPlaygroundFormData } from './inbox-playground';
```
Remove if not used

iampearceman replied:
okay.

scopsy commented on apps/dashboard/src/components/activity/components/step-progress-bar.tsx:
What is this component? Was it part of the scope?

iampearceman replied:
No. Deleted :)

iampearceman committed:
Merge branch 'next' into onboarding-emil

iampearceman committed:
Addressed all the provided comments

iampearceman committed:
Added some more UI fixes

iampearceman committed:
Few more fixes

scopsy committed:
Merge branch 'next' into onboarding-emil

scopsy committed:
Add reload support for user and organization in inbox page

Introduces a reload method to the SelfHostedUser type and updates the InboxUsecasePage to reload user and organization data when environments change. Also adds a refetchInterval option to useFetchEnvironments for periodic environment updates.

scopsy committed:
Add showError option to useFetchEnvironments hook

Introduces a showError parameter to the useFetchEnvironments hook and updates its usage in EnvironmentProvider and InboxUsecasePage to suppress error display in specific contexts. Also includes minor code cleanup and a fix for optional chaining in inbox-playground.

scopsy committed:
Merge branch 'next' into onboarding-emil
